hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -289,6 +290,7 @@ protected boolean isUpdateRecord(HoodieRecord<T> hoodieRecord) {
 
   private void bufferRecord(HoodieRecord<T> hoodieRecord) {
     Schema schema = useWriterSchema ? writeSchemaWithMetaFields : writeSchema;
+    DeleteContext recordDeleteContext = useWriterSchema ? deleteContextWithMetaFields : deleteContext;
```

Comment:
the meta fields are not there for in-memory records in write path, so the `DeleteContext#withReaderSchema` does not really work. So just maintain one singleton `DeleteContext` in the write handle, and within `HoodieRecord.checkIsDelete`, handle the `HoodieOperation` specifically because it is already there in the record.